lib/symlink/index.js

```diff
@@ -51,22 +51,24 @@ function symlink(outFolder, opt) {
       srcPath = path.relative(file.base, srcPath);
     }
 
-    // TODO: make DRY with .dest()
-    var dirMode = number(opt.dirMode, file);
-    var writeFolder = path.dirname(file.path);
-
-    fo.mkdirp(writeFolder, dirMode, onMkdirp);
+    // Because fs.symlink does not allow atomic overwrite option with flags, we
+    // delete and recreate if the link already exists and overwrite is true.
+    if (file.flag === 'w') {
+      // TODO What happens when we call unlink with windows junctions?
```

Comment:
I'm not too familiar with node source, but that block looks like a special case for dangling links, rather than junction links?

But again, I am utterly useless on Windows -- would really love for someone to just try this out (create junction same way as vinyl-fs, and then see what happens after `unlink`)...